doc/Doxyfile

```diff
@@ -746,7 +746,7 @@ WARN_IF_DOC_ERROR      = YES
 # parameter documentation, but not about the absence of documentation.
 # The default value is: NO.
 
-WARN_NO_PARAMDOC       = YES
+WARN_NO_PARAMDOC       = NO
```

Comment:
@hcho3 what's your take on this? Do we want to enforce better documentation for arguments?

Reply:
We should. But for some reason this didn't take effect in old versions of Doxygen and most types in dmlc-core do not document their parameters.
Thus tests fail with newer versions of doxygen (as we treat warnings as errors)

Reply:
So a separate effort is needed to document all parameters and reenable this